ompi/mca/fs/ufs/fs_ufs_file_open.c

```diff
@@ -105,12 +106,6 @@ mca_fs_ufs_file_open (struct ompi_communicator_t *comm,
                    component. */
                 fh->f_flags |= OMPIO_LOCK_ENTIRE_FILE;
             }
-            else {
-                fh->f_flags |= OMPIO_LOCK_NEVER;
-            }
-        }
-        else {
-            fh->f_flags |= OMPIO_LOCK_NEVER;
         }
         free (fstype);
```

Comment:
Not knowing this code path very well, it looks to me like you handled Lustre and NFS just fine.  However, is there any case other than NFS and Lustre?  Seems that previously such a case would have set OMPIO_LOCK_NEVER but now it would be unset (LOCK_RANGES).

Reply:
the fs/ufs component handles everything that is not Lustre at the moment (e.g. NFS, local file systems, BeeGFS, etc.). We treat all of them equal except for NFS. Lustre requires special handling because it allows us to modify some additional parameters (stripe size, stripe count, etc.), hence it has its own component. So from the fs/ufs component perspective its either NFS (in which case we activate locking) or everything else (in which case we now do range locking instead of no locking).

Reply:
Forgot that we also have separate components for GPFS and IME, not sure how much they are in use however. We also used to have separate components for PVFS2/OrangeFS, but that was removed a few weeks ago.

Reply:
I just re-read the commit message.  Your intent is to change those paths from LOCK_NEVER to LOCK_RANGE, which is what you have done.  👍